src/ucs/config/parser.c

```diff
@@ -931,21 +931,47 @@ ucs_status_t ucs_config_parser_set_value(void *opts, ucs_config_field_t *fields,
 }
 
 ucs_status_t ucs_config_parser_get_value(void *opts, ucs_config_field_t *fields,
-                                        const char *name, char *value, size_t max)
+                                         const char *name, char *value,
+                                         size_t max)
 {
-    ucs_config_field_t *field;
-    void *value_ptr;
+    ucs_config_field_t  *field;
+    ucs_config_field_t  *sub_fields;
+    void                *sub_opts;
+    void                *value_ptr;
+    size_t              name_len;
+    ucs_status_t        status;
 
-    for (field = fields; field->name; ++field) {
-        //TODO table
-       if (!strcmp(field->name, name)) {
-           value_ptr = (char*)opts + field->offset;
-           field->parser.write(value, max, value_ptr, field->parser.arg);
-           return UCS_OK;
-       }
+    if (!opts || !fields || !name || (!value && (max > 0))) {
+        return UCS_ERR_INVALID_PARAM;
     }
 
-    return UCS_ERR_INVALID_PARAM;
+    for (field = fields, status = UCS_ERR_NO_ELEM;
+         field->name && (status == UCS_ERR_NO_ELEM); ++field) {
+
+        name_len = strlen(field->name);
+
+        ucs_debug("compare name \"%s\" with field \"%s\" which is%s subtable",
```

Comment:
this would produce a lot of output in normal execution, need to make it ucs_trace then